src/main/xerus/monstercat/api/Player.kt

```diff
@@ -164,13 +176,44 @@ object Player: FadingHBox(true, targetHeight = 25) {
 		}
 	}
 	
-	private val pauseButton = ToggleButton().id("play-pause").onClick { if(isSelected) player?.pause() else player?.play() }
-	private val stopButton = buttonWithId("stop") { reset() }
-	private val volumeSlider = Slider(0.0, 1.0, Settings.PLAYERVOLUME()).scrollable(0.05).apply {
-		prefWidth = 100.0
-		valueProperty().listen { updateVolume() }
-	}
-	
+	private val pauseButton = ToggleButton().id("play-pause")
+			.tooltip("Pause / Play")
+			.onClick { if (isSelected) player?.pause() else player?.play() }
+	private val stopButton = Button().id("stop")
+			.tooltip("Stop playing")
+			.onClick {
+				reset()
+				Playlist.clear()
+			}
+	private val volumeSlider = Slider(0.0, 1.0, Settings.PLAYERVOLUME())
+			.scrollable(0.05)
+			.apply {
+				prefWidth = 100.0
+				valueProperty().listen { updateVolume() }
+				tooltip = Tooltip("Volume")
+			}
+	private val shuffleButton = ToggleButton().id("shuffle")
+			.tooltip("Shuffle")
+			.apply { selectedProperty().bindBidirectional(Playlist.shuffle) }
+	private val repeatButton = ToggleButton().id("repeat")
+			.tooltip("Repeat all")
+			.apply { selectedProperty().bindBidirectional(Playlist.repeat) }
+	private val skipbackButton = buttonWithId("skipback") { playPrev() }
+			.tooltip("Previous")
+			.apply { Playlist.currentIndex.listen { value ->
+				isDisable = value == 0
+				isVisible = value != 0
+			} }
+	private val skipButton = buttonWithId("skip") { playNext() }
+			.tooltip("Next")
+			.apply {
+				arrayOf<Observable>(Playlist.currentIndex, Playlist.repeat, Playlist.shuffle).addListener {
+					val disabled = Playlist.currentIndex.value == Playlist.tracks.lastIndex && !Playlist.repeat.value && !Playlist.shuffle.value
+					isDisable = disabled
+					isVisible = !disabled
```

Comment:
either disable or make invisible. No need to do both

Reply:
Yeah, just checked the doc, invisible nodes doesn't receive events. So I'm good to go with only isVisible

Reply:
a6e0ae4813ac9e88a69c7d641ab6233e8d955e4a

Reply:
a6e0ae4813ac9e88a69c7d641ab6233e8d955e4a